src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngineConfig.java

```diff
@@ -37,9 +39,30 @@
     "This value is ignored if the dataset or temporary bucket already exist.")
   protected String location;
 
+  @Name(NAME_RETAIN_TABLES)
+  @Macro
+  @Nullable
+  @Description("Select this option if the pipeline should retain all temporary BigQuery tables created during the " +
+    "execution of the pipeline.")
+  protected Boolean retainTables;
+
+  @Name(NAME_TEMP_TABLE_TTL_HOURS)
+  @Macro
+  @Nullable
+  @Description("Set table TTL for temporary BigQuery tables, in number of hours. Tables will be deleted " +
+    "automatically on pipeline completion.")
+  protected Integer tempTableTTLHours;
+
   @Nullable
   public String getLocation() {
     return location;
   }
 
+  public Boolean getRetainTables() {
```

Comment:
nit: for readability, name boolean getters with 'is' or 'should' instead of 'get'. 'shouldRetainTables()'

Reply:
Updated.